assets/app/scripts/app.js

```diff
@@ -17,24 +18,121 @@ angular
     'ngSanitize',
     'ngTouch'
   ])
-  .config(function ($routeProvider, $locationProvider) {
+  // configure our tabs and routing
+  .config(['$routeProvider', 'HawtioNavBuilderProvider', function($routeProvider, builder) {
+    var template = function() {
+      // TODO - Don't love triggering the show/hide drawer here, would prefer if 
+      // we could listen for an event that the nav was being redrawn and 
+      // check HawtioNav.selected()
+      if (this.isSelected()) {
+        if (this.tabs && this.tabs.length > 0) {
+          $("body").addClass("show-drawer");
+        }
+        else {
+          $("body").removeClass("show-drawer");
+        }
+      }
+      return "<sidebar-nav-item></sidebar-nav-item>";        
+    };
+
+    var projectHref = function(path) {
+      return function() {
+        var injector = HawtioCore.injector;
+        if (injector) {
+          var routeParams = injector.get("$routeParams");
+          if (routeParams.project) {
+            return "/project/" + routeParams.project + "/" + path;
+          }
+        }
+        return "/project/:project/" + path; 
+      }
+    };
+
+    // TODO - hawtio nav runs the href() function once during initial
+    // build and then uses that in the isSelected function it creates
+    // we need to always get the current href which is dependent on 
+    // the current project.  Want to propose upstream fix to always
+    // run item.href()
+    var isSelected = function() {
+      var injector = HawtioCore.injector;
+      if (injector) {
+        var location = injector.get("$location");
+      }
+      return location.path().indexOf(this.href()) === 0;
```

Comment:
Yes, that will need to be fixed upstream as well.  @gashcrumb see both my comment in the code about needing to always run item.href() as well as @liggitt's comment about the potential for multiple matches.